semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/Scalacp.scala

```diff
@@ -66,12 +38,12 @@ final class Scalacp private (
 
 object Scalacp {
   def parse(
-      classfile: ToplevelClassfile,
-      settings: Settings,
+      node: ClassNode,
+      scalaSig: ScalaSig,
```

Comment:
I like the new way to organize code, but it's still unsatisfying that we have to pass two things here (`node` and `scalaSig`) instead of one, especially since the latter is computed from the former.

Here's what I propose. How about `node.scalaSig` returns `ScalaSigAttribute`, which is in turn changed to also capture `ClassReader.getClassName`?

Reply:
Done, I introduced a new `case class ScalaSigNode(relativeUri: String, scalaSig: ScalaSig)` to match `ClassNode` and since `ScalaSigAttribute` is already used.